openscm/adapters/dice.py

```diff
@@ -0,0 +1,388 @@
+"""
+Adapter for the climate component from the Dynamic Integrated Climate-Economy (DICE)
+model by William Nordhaus.
+
+Original source: https://sites.google.com/site/williamdnordhaus/dice-rice
+"""
+
+from collections import namedtuple
+from datetime import datetime
+from math import log2
+from typing import Any
+
+import numpy as np
+
+from ..adapter import Adapter
+from ..parameters import ParameterType
+from ..timeseries_converter import (
+    ExtrapolationType,
+    InterpolationType,
+    create_time_points,
+)
+
+YEAR = 365 * 24 * 60 * 60
+
+MODEL_PARAMETER_DEFAULTS = {
+    # Initial concentration in atmosphere
+    "M_atm0": (830.4, "GtC"),
+    # Equilibrium concentration atmosphere
+    "M_atm_eq": (588, "GtC"),
+    # Lower bound concentration in atmosphere
+    "M_atm_lower": (10, "GtC"),
+    # Initial concentration in lower strata
```

Comment:
```suggestion
    # Initial concentration in lower stratosphere
```

Reply:
similarly below, I think using shortenings here will just be confusing

Reply:
hmm, what about putting pur comments and then only quote the original code comments? would want to have them in there for people to easily compare to the original dice code

Reply:
ah ok that's sensible then. I'm wondering if we should have a disclaimer at the top that says something like, "In the following we stick as closely as possible to the original dice code. We then use the adapter to translate the DICE conventions into OpenSCM conventions, effectively resolving any confusing DICE internal conventions in the process for any OpenSCM users."